applications/webapp/src/Pages/User/components/detailHolder.tsx

```diff
@@ -10,65 +10,98 @@
  * LICENSE file in the root directory of this source tree.
  */
 
-import React from "react";
+import React, { useState } from "react";
 import {
   Card,
   CardContent,
   FormControl,
   Grid,
   useTheme,
   TextField,
+  Button,
+  Box,
 } from "@mui/material";
 import { styles } from "../styles/style";
-
-interface DetailHolderProps {
-  firstName: string;
-  lastName: string;
-  email: string;
-  phoneNumber: string;
-  address: string;
-  birthdate: string;
-}
+import { useMutation } from "@apollo/client";
+import { EditUserMutation } from "../graphql/mutation";
+import { DetailHolderProps } from "../../../Pages/interface";
 
 export default function DetailHolder({
-  firstName,
-  lastName,
-  email,
-  phoneNumber,
-  address,
-  birthdate,
+  user,
+  card,
 }: Readonly<DetailHolderProps>) {
   const theme = useTheme();
   const style = styles(theme).detailHolder;
+
+  const [firstname, setFirstName] = useState(user.firstName);
+  const [lastname, setLastName] = useState(user.lastName);
+
+  const [editUser] = useMutation(EditUserMutation);
+
+  const userInput = {
+    username: user.username,
+    firstName: firstname,
+    lastName: lastname,
+  };
+  const updateUser = async () => {
+    const zeroElevation = 0;
+    try {
+      const { data } = await editUser({
+        variables: {
+          userInput,
+        },
+      });
+      console.log("User:", data.editUser);
+    } catch (error) {
+      console.error("Error:", error);
```

Comment:
might work for now, not overall logging errors to console as well as logging anything to console in web app is a very bad practice. needs a better way.